### PR TITLE
feat(issue-41): set and inherit labels

### PR DESCRIPTION
**Pull request by lampajr**
**Thank you for submitting this pull request**

fix  https://github.com/kiegroup/git-backporting/issues/41

This change extends the functionality letting user to 
- Set their own `labels` to the newly created backporting one
- Inherit the labels from the original pull request
- Both previous options together

### Checklist
- [x] Tests added if applicable.
- [x] Documentation updated if applicable.

> **Note:** `dist/cli/index.js` and `dist/gha/index.js` are automatically generated by git hooks and gh workflows.

### CODE REVIEW

**Comment by lampajr** [comment]
In addition to the enhancement itself, I improved a bit the test coverage, here the results right now:
```
----------------------------|---------|----------|---------|---------|-------------------------------
File                        | % Stmts | % Branch | % Funcs | % Lines | Uncovered Line #s             
----------------------------|---------|----------|---------|---------|-------------------------------
All files                   |   90.25 |       80 |   89.28 |   90.39 |                               
 args                       |   95.83 |     92.3 |     100 |   95.65 |                               
  args-parser.ts            |   83.33 |       80 |     100 |   83.33 | 21                            
  args-utils.ts             |     100 |      100 |     100 |     100 |                               
 args/cli                   |     100 |      100 |     100 |     100 |                               
  cli-args-parser.ts        |     100 |      100 |     100 |     100 |                               
 args/gha                   |     100 |      100 |     100 |     100 |                               
  gha-args-parser.ts        |     100 |      100 |     100 |     100 |                               
 configs                    |     100 |      100 |     100 |     100 |                               
  configs-parser.ts         |     100 |      100 |     100 |     100 |                               
 configs/pullrequest        |    91.3 |    96.29 |     100 |    91.3 |                               
  pr-configs-parser.ts      |    91.3 |    96.29 |     100 |    91.3 | 22-23                         
 git                        |    82.5 |    53.57 |   76.47 |   82.05 |                               
  git-cli.ts                |    62.5 |    21.42 |      60 |    62.5 | 38-43,63-67,88-89,109-110,124 
  git-client-factory.ts     |    90.9 |       60 |     100 |    90.9 | 18,46                         
  git-util.ts               |     100 |      100 |     100 |     100 |                               
  git.types.ts              |     100 |      100 |     100 |     100 |                               
 git/github                 |   91.07 |    84.61 |   88.23 |   92.59 |                               
  github-client.ts          |   91.42 |       75 |     100 |   91.42 | 66,91,104                     
  github-mapper.ts          |   84.61 |     87.5 |   77.77 |    90.9 | 28                            
  octokit-factory.ts        |     100 |      100 |     100 |     100 |                               
 git/gitlab                 |   95.18 |    65.21 |     100 |   95.18 |                               
  gitlab-client.ts          |   96.82 |       75 |     100 |   96.82 | 147,151                       
  gitlab-mapper.ts          |      90 |       60 |     100 |      90 | 23,78                         
 logger                     |   88.88 |      100 |   77.77 |   88.88 |                               
  console-logger-service.ts |      90 |      100 |   83.33 |      90 | 15                            
  logger-service-factory.ts |     100 |      100 |     100 |     100 |                               
  logger.ts                 |   66.66 |      100 |      50 |   66.66 | 13                            
 runner                     |   84.78 |      100 |   66.66 |   84.78 |                               
  runner.ts                 |   84.78 |      100 |   66.66 |   84.78 | 30-40                         
----------------------------|---------|----------|---------|---------|-------------------------------
Test Suites: 14 passed, 14 total
Tests:       103 passed, 103 total
Snapshots:   0 total
```

**Review by shubhbapna — CHANGES_REQUESTED** [review]

File: src/service/args/args-utils.ts [hunk 1]
```diff
@@ -19,4 +19,35 @@ export function parseArgs(configFileContent: string): Args {
...
+ * @returns the value or undefined
+ */
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export function getOrUndefined(value: string): string | undefined {
+  return value !== "" ? value : undefined;
+}
+
+// get rid of inner spaces too
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export function getAsCleanedCommaSeparatedList(value: any): string[] | undefined {
```
[the remark this answers is not in the document]
Comment (lampajr) [reply src/service/args/args-utils.ts:36]: leftover, thx for spotting this

File: src/service/git/github/github-client.ts [hunk 2]
```diff
@@ -59,13 +59,26 @@ export default class GitHubClient implements GitClient {
...
+          owner: backport.owner,
+          repo: backport.repo,
+          issue_number: (data as PullRequest).number,
+          labels: backport.labels,
+        });
+      } catch (error) {
+        this.logger.error(`Error setting labels: ${error}`);
+      }
+    }
+
```
[the remark this answers is not in the document]
Comment (lampajr) [reply src/service/git/github/github-client.ts:81]: I definitely agree with you, I was not thinking about that.
I would open an issue for this and do that in a separate PR if you agree.

[the remark this answers is not in the document]
Comment (shubhbapna) [reply src/service/git/github/github-client.ts:81]: Yep sounds good!

**Comment by shubhbapna** [comment]
Also for the code coverage we can enable sonar cloud for the project so that for each PR we can view the coverage info on sonar cloud automatically. Wdyt?

File: src/service/git/github/github-client.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (shubhbapna) [reply src/service/git/github/github-client.ts:81]: issue created: https://github.com/kiegroup/git-backporting/issues/49

**Review by shubhbapna — APPROVED** [review]

File: src/service/git/github/github-client.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (lampajr) [reply src/service/git/github/github-client.ts:81]: thanks!

**Comment by lampajr** [comment]
> Also for the code coverage we can enable sonar cloud for the project so that for each PR we can view the coverage info on sonar cloud automatically. Wdyt?

Yeah that is for sure a great improvement, created https://github.com/kiegroup/git-backporting/issues/50 to keep track of this :)

**Review by Ginxo — APPROVED** [review]
LGTM, just package.json version should be upgraded right? also CHANGELOG...
many thanks for this :heart:

**Comment by lampajr** [comment]
> LGTM, just package.json version should be upgraded right? also CHANGELOG... many thanks for this heart

I created this workflow [1] to automatically update (based on diff commits) both package.json and CHANGELOG such that we do not have to do it manually. I already added **release** section [2] in the README. 

[1] https://github.com/kiegroup/git-backporting/actions/workflows/prepare-release.yml
[2] https://github.com/kiegroup/git-backporting/#release

**Pull request merged**